python/ray/data/tests/test_dataset.py

```diff
@@ -4258,6 +4258,25 @@ def test_func(block):
     wait_for_condition(lambda: (ray.available_resources().get("CPU", 0) == cpus))
 
 
+def test_actorpoolstrategy_default_max_actors(shutdown_only):
+    def f(x):
+        import time
+
+        time.sleep(1)
+        return x
+
+    num_cpus = 5
+    ray.init(num_cpus=num_cpus)
+    compute_strategy = ray.data.ActorPoolStrategy()
+    ray.data.range(10, parallelism=10).map_batches(f, compute=compute_strategy)
+    expected_max_num_workers = math.ceil(
+        num_cpus * (1 / compute_strategy.ready_to_total_workers_ratio)
+    )
+    assert (
+        compute_strategy.num_workers <= expected_max_num_workers
```

Comment:
Shall we also assert that it scaled up to at least 5 workers?

Reply:
@ericl - makes sense, added.